Add example for 'podman play kube'

YaSuenag opened the pull request:
# Pull Request

## Summary

I'd like to propose to add example for CASDK demonstration.

This example provides both the SDK and Swagger UI with just one command. Thus demonstrator does not need to prepare each components for WebAPI. I used this in [GSF Meetup Tokyo](https://github.com/Green-Software-Foundation/meetup/issues/49), then it works fine!

This example would use CASDK WebAPI container image tagged `pre` from GitHub Packages.

## Changes

- samples/casdk-demo (new)

## Checklist

- [ ] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [x] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [x] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?

No

## Is this a breaking change?

No

> Please follow
> [GitHub's suggested syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
> to link Pull Requests to Issues via keywords

This PR Closes #<issue_number>

YaSuenag commented:
I'd like to contribute this example because this manifest would be useful to demonstrate the SDK with just single command on Linux box.

danuw commented:
I think @Willmish already has podman but otherwise happy to install next week and review.

danuw commented:
@Willmish assigned this to you for now as I know you have podman and can probably review this quickly... let me know otherwise

Willmish commented:
#417 @Willmish to look at for next week

danuw commented:
@Willmish any news please?

danuw commented:
@YaSuenag let me know if you have a few minutes to go over this at some point. 
Also, ideally make the update so we have a settings template that does not get tracked.

Sophietn commented:
@danuw FYI @YaSuenag cannot see your comments, so is unable to review. 
@danuw please resubmit your comments to unblock review

danuw reviewed (changes requested):
sorry indeed I forgot to commit
@YaSuenag it should show now

YaSuenag commented:
Thanks @danuw for your comment! I updated this PR.

* This PR is no longer to have `casdk-config.yaml` - it would be auto-generated from environment variables. See updated README: https://github.com/YaSuenag/carbon-aware-sdk/tree/podman-example/samples/casdk-demo#example
* The error what you saw might be caused by SELinux. I added `demo.sh` to start/stop the demonstration - it would work whether SELinux is enabled or not. Can you test `./demo.sh start` to run?
    * You can stop via `./demo.sh stop`. See [README.md](https://github.com/YaSuenag/carbon-aware-sdk/tree/podman-example/samples/casdk-demo#how-to-run) for details.

YaSuenag commented:
This PR have to be aware of exposed port from Web API. So I want to tweak this PR after .NET 8 migration.

YaSuenag commented:
I updated this PR to refer new (.NET 8 migrated) `pre` WebAPI container. Please review! @danuw @Willmish

danuw reviewed (approved):
Thank you for those changes - can we get this reproduced once on a different machine (will try but need to double check I have podman) and then we can get this through.

@YaSuenag

YaSuenag commented:
It works on my environment:

Fedora 40
![スクリーンショット 2024-06-04 171256](https://github.com/Green-Software-Foundation/carbon-aware-sdk/assets/7421132/82a1acad-8cb9-46ce-9918-a6b49d6a7b6d)

Ubuntu 24.04 (WSL2)
![スクリーンショット 2024-06-04 171856](https://github.com/Green-Software-Foundation/carbon-aware-sdk/assets/7421132/dd41f597-6d04-4e8a-ac34-1f99820a7274)

tiwatsuka commented:
I also tried. It worked as expected in my environment.

![image](https://github.com/Green-Software-Foundation/carbon-aware-sdk/assets/12785901/2668902d-6379-4d7a-bb71-f29fccd573a0)
